src/source/Signaling/LwsApiCalls.c

```diff
@@ -93,10 +101,45 @@ INT32 lwsHttpCallbackRoutine(struct lws* wsi, enum lws_callback_reasons reason,
 
         case LWS_CALLBACK_ESTABLISHED_CLIENT_HTTP:
             status = lws_http_client_http_response(wsi);
-            DLOGD("Connected with server response: %d", status);
+            getStateMachineCurrentState(pSignalingClient->pStateMachine, &pStateMachineState);
 
+            DLOGD("Connected with server response: %d", status);
             pLwsCallInfo->callInfo.callResult = getServiceCallResultFromHttpStatus((UINT32) status);
 
+            lws_hdr_copy(wsi, &dateHdrBuffer[0], MAX_DATE_HEADER_BUFFER_LENGTH, WSI_TOKEN_HTTP_DATE);
```

Comment:
What if the copy fails? I believe this call returns length if success and -1 if it fails. Can we do this instead?

`len = lws_hdr_copy(wsi, &dateHdrBuffer[0], MAX_DATE_HEADER_BUFFER_LENGTH, WSI_TOKEN_HTTP_DATE);`. This will knock out line 113 requirement. Correct me if I misunderstood this.

Reply:
I copied the usage from a sample or a test in lws, I can't find it now, this is similar:

https://github.com/warmcat/libwebsockets/blob/0e9842a0c0e4ab3d0bbf63536d7f9f4b5c3e7c8b/lib/roles/http/date.c#L161

Let me look more into this I'm not 100% sure yet if the return value of `lws_hdr_copy` is the same as what `lws-hdr_total_length` returns

Reply:
I looked at the implmenetation and yes `lws_hdr_copy` returns exactly the same thing as `lws_hdr_total_length` so I'll make that simplification